lib/bloc/account/add_funds_bloc.dart

```diff
@@ -1,29 +1,107 @@
-
 import 'dart:async';
+import 'dart:convert';
+
 import 'package:breez/bloc/account/account_model.dart';
+import 'package:breez/logger.dart';
 import 'package:breez/services/breezlib/breez_bridge.dart';
 import 'package:breez/services/injector.dart';
+import 'package:flutter/services.dart';
+import 'package:http/http.dart' as http;
+import "package:ini/ini.dart";
+import 'package:rxdart/rxdart.dart';
+
+import 'account_model.dart';
+import 'moonpay_order.dart';
 
 class AddFundsBloc {
+  static const String ACCOUNT_SETTINGS_PREFERENCES_KEY = "account_settings";
+  static const String PENDING_MOONPAY_ORDER_KEY = "pending_moonpay_order";
+  String _moonPayURL;
+
   final _addFundRequestController = new StreamController<void>();
+
   Sink<void> get addFundRequestSink => _addFundRequestController.sink;
 
   final _addFundResponseController = new StreamController<AddFundResponse>();
+
   Stream<AddFundResponse> get addFundResponseStream => _addFundResponseController.stream;
 
-  AddFundsBloc(String userID){
-    BreezBridge breezLib = new ServiceInjector().breezBridge;
-    _addFundRequestController.stream.listen((request){
+  final _moonPayOrderController = new BehaviorSubject<MoonpayOrder>();
+
+  Stream<MoonpayOrder> get moonPayOrderStream => _moonPayOrderController.stream;
+
+  Sink<MoonpayOrder> get moonPayOrderSink => _moonPayOrderController.sink;
+
+  AddFundsBloc(String userID) {
+    ServiceInjector injector = ServiceInjector();
+    BreezBridge breezLib = injector.breezBridge;
+    _addFundRequestController.stream.listen((request) {
       _addFundResponseController.add(null);
-        breezLib.addFundsInit(userID)
+      breezLib
+          .addFundsInit(userID)
           .then((reply) => _addFundResponseController.add(new AddFundResponse(reply)))
           .catchError(_addFundResponseController.addError);
-      })
-      .onDone(_dispose);
+    }).onDone(_dispose);
+    _listenAccountSettings(injector);
+    _handleMoonpayOrders(injector);
+  }
+
+  String get moonPayURL => _moonPayURL;
+
+  _listenAccountSettings(ServiceInjector injector) async {
+    var preferences = await injector.sharedPreferences;
+    var accountSettings = preferences.getString(ACCOUNT_SETTINGS_PREFERENCES_KEY);
```

Comment:
Ideally this value should be saved at the same place it is read from.
I would move the save action from the AccountBloc to the AddFundsBloc.